proposals/2023-08-21-utf8.md

```diff
@@ -0,0 +1,245 @@
+  * **Exposition Format**: Characters are escaped with a custom backslash format.
+
+### What will the new syntax look like?
+
+**tl;dr:**
+
+`{"my.noncompliant.metric", "label"="value"}`
+
+#### Oh no, I hate it
+
+A full discussion of rejected alternatives can be found below, but a common first reaction is, why move the metric name inside the braces? Why not leave it outside, like: `"my.noncompliant.metric"{"label"="value"}`.
+
+There are some specific risks with this format, but we also wanted to weigh the value of a syntax that is **more familiar to current experts** vs **more approachable for new users**. We take the optimistic position that there are more potential new users of Prometheus than there are existing users, and therefore decided to give more weight to that framing.
+
+The previous syntax makes sense because it resembles a function call, the way that `foo{bar}` is like `foo(bar)`. The metric name is the primary selector, and the braces enclose parameters on that selection. If we were to add the additional quote characters, this frame of reference is lost. For new users, visually parsing the query, with brace in the middle of a noisy series of quoted values, is likely to be confusing.
+
+Instead, we believe that it is more intuitive and easier to teach new users to think of the brace as **the characters that surround the parameters for selecting data**, where **the "metric name" is a special case** of that kind of selector. This is similar to the approach chosen for [LogQL](https://grafana.com/docs/loki/latest/query/) and [TraceQL](https://grafana.com/docs/tempo/latest/traceql/), which do not have a special case "metric name" equivalent and thus have no items outside the curly braces.
+
+Furthermore, PromQL users will already know that it's possible to put the metric name inside the braces, for the purpose of using an operator to enable richer metric selection, like: `{__name__=~"my\.metric.*"}`. The chosen syntax serves as a shortening of this operation, and is forward-compatible with future metric matching prefix operators if we decide to implement them.
+
+### Syntax Changes
+
+We propose the following change in syntax to PromQL and the exposition format:
+
+* Label names may be quoted, either double quotes for the exposition format, or singles/doubles/backticks for PromQL.
+* A quoted string inside the label set (`{}`) without a label matching operator is default-assigned to the label name `__name__` (the metric name).
+* Two strings without a label matching operator is a syntax error.
+* Style prefers, but does not enforce, that the string without a label matching operator is the first term in the label set.
+* Strings in quotes will be unquoted using the mechanism for each language: strconv.Unquote for PromQL, and the OpenMetrics escaping mechanism for the exposition format.
+* In the exposition format, the TYPE, HELP, and UNIT lines can have double quotes around the metric name and similarly use Go quoting.
```

Comment:
I think we should keep the quoting here consistent with the other quoting rules in OM (rather than suddenly introducing Go quoting). So this should also be "the OpenMetrics escaping mechanism".